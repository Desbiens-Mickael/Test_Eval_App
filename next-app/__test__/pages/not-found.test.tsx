import NotFound from "@/app/not-found";
import { render, screen } from "@testing-library/react";

// Render the NotFound component before each test case
beforeEach(() => {
  render(<NotFound />);
});

describe("NotFound component", () => {
  it("should render a 404 page", () => {
    const h1 = screen.getByRole("heading", { level: 1 });
    expect(h1).toBeInTheDocument();
    expect(h1).toHaveTextContent("Not Found");
  });

  it("should render an information text", () => {
    const text = screen.getByRole("paragraph");
    expect(text).toBeInTheDocument();
    expect(text).toHaveTextContent("Impossible de trouver la ressource demandée");
  });

  it("should render an image in the 404 page", () => {
    const image = screen.getByAltText("not found");
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("src", "/assets/images/not-found.png");
  });

  it("should render a link to the home page", () => {
    const link = screen.getByRole("link", { name: "Retour à l'accueil" });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "/");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });
});
