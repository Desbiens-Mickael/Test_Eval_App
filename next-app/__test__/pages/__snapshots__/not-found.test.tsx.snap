// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-full w-full flex flex-col justify-center items-center p-4"
  >
    <h1
      class="text-primary text-6xl font-bold"
    >
      Not Found
    </h1>
    <img
      alt="not found"
      height="500"
      src="/assets/images/not-found.png"
      width="500"
    />
    <p
      class="text-primary text-2xl md:text-3xl lg:text-4xl font-bold text-center"
    >
      Impossible de trouver la ressource demandée
    </p>
    <a
      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 mt-5"
      href="/"
    >
      Retour à l'accueil
    </a>
  </div>
</DocumentFragment>
`;
